educe/stac/edit/cmd/delete_anno.py

```diff
@@ -0,0 +1,112 @@
+# Author: Eric Kow
+# License: CeCILL-B (French BSD3-like)
+
+"""Delete an annotation"""
+
+from __future__ import print_function
+import sys
+
+from educe.stac.util.args import\
+    add_usual_input_args, add_usual_output_args,\
+    read_corpus, get_output_dir, announce_output_dir,\
+    anno_id
+from educe.stac.util.glozz import anno_id_from_tuple, anno_id_to_tuple
+from educe.stac.util.output import save_document
+
+
+
+def _is_match(wanted):
+    """
+    Given an annotation id, return a predicate that checks if
+    an annotation id matches
+    """
+    def pred(anno):
+        "curried second arg"
+        return anno_id_to_tuple(anno.local_id()) == wanted
+    return pred
+
+
+def _delete_in_doc(del_id, doc):
+    """Delete the annotations with the given id in the given document
+
+    NB: modifies doc
+    """
+    pretty_id = anno_id_from_tuple(del_id)
+    is_ok = lambda x: not _is_match(del_id)(x)
+    matches = [x for x in doc.annotations() if not is_ok(x)]
```

Comment:
Yup, that's a lot better.

So you can imagine how this came about, right?
There used to be a filter function I was trying to use.
But these days am trying to go with the flow more.
Just didn't notice I could clean out is_match along the way


On 21 May 2015 at 11:59, Mathieu Morey <notifications@github.com> wrote:

> In educe/stac/edit/cmd/delete_anno.py
> <https://github.com/irit-melodi/educe/pull/49#discussion_r30788064>:
>
> > +    an annotation id matches
> > +    """
> > +    def pred(anno):
> > +        "curried second arg"
> > +        return anno_id_to_tuple(anno.local_id()) == wanted
> > +    return pred
> > +
> > +
> > +def _delete_in_doc(del_id, doc):
> > +    """Delete the annotations with the given id in the given document
> > +
> > +    NB: modifies doc
> > +    """
> > +    pretty_id = anno_id_from_tuple(del_id)
> > +    is_ok = lambda x: not _is_match(del_id)(x)
> > +    matches = [x for x in doc.annotations() if not is_ok(x)]
>
> What about
>
> is_ok = lambda x: anno_id_to_tuple(x.local_id()) != wanted
>
> then?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/irit-melodi/educe/pull/49/files#r30788064>.
>



-- 
Eric Kow <http://erickow.com>